src/handler.rs

```diff
@@ -273,17 +272,20 @@ impl RequestHandler {
             {
                 Ok(result) => (result.resp, result.is_precompressed, Some(result.file_path)),
                 Err(status) => {
-                    let mut resp = None;
+                    let resp = None;
 
                     // Check for a fallback response
                     #[cfg(feature = "fallback-page")]
-                    if req.method().is_get()
+                    let resp = if req.method().is_get()
                         && status == StatusCode::NOT_FOUND
                         && !self.opts.page_fallback.is_empty()
                     {
-                        resp = Some(fallback_page::fallback_response(&self.opts.page_fallback));
-                    }
+                        Some(fallback_page::fallback_response(&self.opts.page_fallback))
+                    } else {
+                        resp
+                    };
 
+                    #[allow(clippy::unnecessary_literal_unwrap)]
```

Comment:
I wonder if you even need this. Did you get lint errors without it? I bet that it compiles just fine.

Reply:
Yes, clipply complains when run with `--no-default-features`. It still compiles of course, but I just don’t want the noise.

Reply:
Strange because for me `cargo build --no-default-features` builds with no clippy errors.

Reply:
Btw `cargo build --no-default-features` could be a good addition to the CI `test` matrix in [`devel.yaml`](https://github.com/static-web-server/static-web-server/blob/master/.github/workflows/devel.yml#L247).
Since I want to improve CI anyway, I will take care of this.

Reply:
Tested on nightly but I do not see the error. Can we remove this for now? If fails then we can fix it later. Because for stable compiles just fine.